packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,83 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
```

Comment:
Why does it return `unknown`?

Reply:
When I wrote this, the idea was that the user configuration should be strongly typed only after we've validated that it matches the schema.

So this method just returns the data from the user config without it being validated. That's what we do in `loadConfigAsync`. We can just change this, but I think the `unknown` reflects the type better here.